File: code_samples/lua/oop_interfaces.lua
-- Oop: Interfaces
-- AI/ML Training Sample

Interfaces = {}
Interfaces.__ index = Interfaces

function Interfaces:new()
    local obj = {
        data = ""
    }
    setmetatable(obj, self)
    return obj
end

function Interfaces:process(input)
    self.data = input
end

function Interfaces:getData()
    return self.data
end

function Interfaces:validate()
    return #self.data > 0
end

-- Example usage
local instance = Interfaces:new()
instance:process("example")
print("Data: " .. instance:getData())
print("Valid: " .. tostring(instance:validate()))

File: code_samples/scala/oop_interfaces.scala
/**
 * Oop: Interfaces
 * AI/ML Training Sample
 */

class Interfaces {
  private var data: String = ""
  
  def process(input: String): Unit = {
    data = input
  }
  
  def getData: String = data
  
  def validate: Boolean = data.nonEmpty
}

object InterfacesExample {
  def main(args: Array[String]): Unit = {
    val instance = new Interfaces()
    instance.process("example")
    println(s"Data: ${instance.getData}")
    println(s"Valid: ${instance.validate}")
  }
}
